convert2rhel/subscription.py

```diff
@@ -244,11 +253,17 @@ def install_rhel_subscription_manager():
     )
     if ret_code:
         loggerinst.critical("Failed to install subscription-manager packages. See the above yum output for details.")
-    else:
-        loggerinst.info("Packages installed:\n%s" % "\n".join(rpms_to_install))
-        pkg_names = get_installed_submgr_pkg_names(rpms_to_install)
-        utils.changed_pkgs_control.track_installed_pkgs(pkg_names)
-        loggerinst.debug("Tracking installed packages: %r" % pkg_names)
+
+    loggerinst.info("\nPackages installed:\n%s" % "\n".join(rpms_to_install))
+
+    # Untrack any packages that are present in the system before the install
+    if rpms_to_untrack and pkg_names:
+        for rpm in rpms_to_untrack:
+            loggerinst.debug("Untracking package: %s" % rpm)
+            pkg_names.remove(rpm)
+
+    loggerinst.debug("Tracking installed packages: %r" % pkg_names)
+    utils.changed_pkgs_control.track_installed_pkgs(pkg_names)
 
 
 def get_installed_submgr_pkg_names(rpm_paths):
```

Comment:
```suggestion
def get_pkg_names_from_rpm_paths(rpm_paths):
```

Please move this function to `pkghandler.py` or `utils.py` as it's generic.

Reply:
Done.